### PR TITLE
bump `gopkg.in/DataDog/dd-trace-go.v1` to v1.61.0

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR bumps `gopkg.in/DataDog/dd-trace-go.v1 ` to v1.61.0, the latest at this time.
Small reminder that, except for serverless usage, this lib is mostly used to drive the internal profiling feature.
A few out of ordinary changes required:
- https://github.com/DataDog/dd-trace-go/pull/2368 (including in v1.61.0) changes the default context propagation order resulting in the changes to `TestExtractorExtract`
- with https://github.com/DataDog/dd-trace-go/releases/tag/v1.59.0, ASM is kind of enabled by default, at least from a build perspective (to allow it to be enabled remotely through remote config). Since we do not need/want this on the agent I added the `datadog.no_waf` build tag to all agent processes to opt out of the ASM build requirements. Serverless on the other hand needs this (and its tests as well).

### CODE REVIEW

File: tasks/build_tags.py [hunk 1]
```diff
@@ -175,33 +190,35 @@
         "system-probe-unit-tests": SYSTEM_PROBE_TAGS.union(UNIT_TEST_TAGS),
```
Comment (chouetz) [inline tasks/build_tags.py:190]:
```suggestion
        "system-probe-unit-tests": SYSTEM_PROBE_TAGS.union(UNIT_TEST_TAGS).difference(UNIT_TEST_EXCLUDE_TAGS),
```
but maybe it's the only place where you don't want to exclude it?

File: tasks/build_tags.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
     "containerd",
     "cri",
     "docker",
+    "datadog.no_waf",
```
Comment (chouetz) [inline tasks/build_tags.py:24]: `datadog.no_waf` is funny for a french speaker 🐶

**Review by chouetz — APPROVED** [review]
Ok if my remark is not valid

File: tasks/build_tags.py (hunk 2 above)
Comment (paulcacheux) [reply tasks/build_tags.py:24]: Woof woof !

**Review by purple4reina — APPROVED** [review]
Difference in serverless extension binary size from this change is negligible.

Size of `datadog-agent` v55 binary: 40288688  
Size of `datadog-agent` binary from this branch: 40319128
Difference: ~0.03MB

**Pull request merged**